modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/fields/modal_content/AddFieldsModalContent.tsx

```diff
@@ -158,49 +214,142 @@ const AddFieldsModalContent = ({
 
 				setSelectedKeys(initialSelectedKeys);
 
-				setFields(updatedFields);
+				setInitialFields(updatedFields);
 			}
 		});
 	}, [savedFDSFields, fdsView]);
 
+	useEffect(() => {
+		if (initialFields) {
+			const filteredItems = filterItems(initialFields, debouncedSearch);
+			setFields(filteredItems);
+
+			if (debouncedSearch) {
+				const filteredKeys = getItemsKeys(filteredItems);
+				setExpandedKeys(filteredKeys);
+			}
+		}
+	}, [initialFields, debouncedSearch]);
+
 	return (
 		<>
 			<ClayModal.Header>
 				{Liferay.Language.get('add-fields')}
 			</ClayModal.Header>
 
-			<ClayModal.Body className="bg-light m-4 p-0">
+			<ClayModal.Body className="pt-0 px-0">
```

Comment:
> I attempted to limit the modal width to size xl, but I lost the full height. Any ideas on how to maintain both full height and max-width?

I kinda like the solution as it is now! Limited internal width in full screen modal seems a nice compromise beween small modal with fixed size and a-bit-too-wide full screen modal. @ugeortiz what do you think?

Reply:
Sadly on wide screens looks ugly 😓 

![image](https://github.com/liferay-frontend/liferay-portal/assets/67901/d278a245-0525-49f9-922f-17d94c364a78)